src/ChargeDriftModels/InactiveLayerChargeDriftModel/InactiveLayerChargeDriftModel.jl

```diff
@@ -0,0 +1,108 @@
+"""
+    struct InactiveLayerChargeDriftModel{T <: SSDFloat} <: AbstractChargeDriftModel{T}
+        
+Charge drift model in which the electrons and holes drift along the electric field.
+There factors are considered in the mobility calculation: ionized impurities, neutral impurities, and acoustic phonon.
+
+Ref: [Dai _et al._ (2023)](https://doi.org/10.1016/j.apradiso.2022.110638)
+
+## Fields
+- `calculate_mobility::Function`: Mobility calculation function
+- `neutral_imp_model::AbstractImpurityDensity{T}`: the neutral impurity density model
+- `bulk_imp_model::AbstractImpurityDensity{T}`: the bulk impurity density model (p-type impurity)
+- `surface_imp_model::AbstractImpurityDensity{T}`: the surface impurity density model (n-type impurity)
+
+## Extra field for constructing the model
+- `temperature::T`: temperature of the crystal (Kelvin)
+"""
+
+struct InactiveLayerChargeDriftModel{T <: SSDFloat} <: AbstractChargeDriftModel{T}
+    calculate_mobility::Function
+	neutral_imp_model::AbstractImpurityDensity{T}
+	bulk_imp_model::AbstractImpurityDensity{T}
+    surface_imp_model::AbstractImpurityDensity{T}
+end
+
+function InactiveLayerChargeDriftModel{T}(
+	temperature::T = T(90), # Kelvin
+	neutral_imp_model::AbstractImpurityDensity{T} = ConstantImpurityDensity{T}(T(1e21)),
+	bulk_imp_model::AbstractImpurityDensity{T} = ConstantImpurityDensity{T}(T(-1e16)),
+    surface_imp_model::AbstractImpurityDensity{T} = ConstantImpurityDensity{T}(T(0))
+	) where {T <: SSDFloat}
+
+	function calculate_mobility(pt::AbstractCoordinatePoint{T}, CC::Type{CC_type}) where {T <: SSDFloat, CC_type <: ChargeCarrier}
+		calculate_mobility_(
+			get_impurity_density(neutral_imp_model, pt),
+			get_impurity_density(bulk_imp_model, pt),
+			get_impurity_density(surface_imp_model, pt),
+			temperature, CC)
+	end
+
+    InactiveLayerChargeDriftModel{T}(calculate_mobility, neutral_imp_model, bulk_imp_model, surface_imp_model)
+end
+
+function calculate_mobility_(
+	Nn::T, bulk_imp::T, surface_imp::T,
+	temperature::T,
+	::Type{Hole}) where {T}
+	Ni = -bulk_imp + surface_imp
+
+	μI = 2.35e19*temperature^1.5/Ni/log(9.13e19*temperature^2/Ni) + 1.51e20*temperature^1.5/Ni/log(5.82e20*temperature^2/Ni)
+	μA = 7.77e3 * temperature^-1.5
+	μN = 1e2/Nn * (2.31e18+2.36e20) * 0.82 * (0.228*temperature^0.5 + 0.976*temperature^-0.5)
+
+	1/(1/μI + 1/μA + 1/μN)
+end
+function calculate_mobility_(
```

Comment:
It's very uncommon to have a function name ending in `_`.
Consider renaming this function and making it more descriptive, e.g.
```julia
_calculate_mobility_from_impurities
```
or something like this.